astropy/io/ascii/ecsv.py

```diff
@@ -170,15 +177,34 @@ def get_cols(self, lines):
         # Transfer attributes from the column descriptor stored in the input
         # header YAML metadata to the new columns to create this table.
         for col in self.cols:
-            for attr in ('description', 'format', 'unit', 'meta'):
+            for attr in ('description', 'format', 'unit', 'meta', 'subtype'):
                 if attr in header_cols[col.name]:
                     setattr(col, attr, header_cols[col.name][attr])
+
             col.dtype = header_cols[col.name]['datatype']
+            if col.dtype not in ECSV_DATATYPES:
```

Comment:
Maybe move this down together with the subtype?

Reply:
I left this here to prevent useless subsequent processing if things are just going to fail.